Add contributing guidelines (#118)
Imageomics/catalog: CONTRIBUTING.md

# Contributing to the Catalog

Thank you for your interest in contributing to the catalog! This document outlines the standards and guidelines for contributing to this template repository.

## Overview

> This catalog is built with JavaScript and relies on API calls to code, data, model, and spaces repositories to populate the web app in real time. It is deployed via GitHub Pages.

Reviewer: ```suggestion
This catalog is built with JavaScript and relies on API calls to code, data, model, and spaces repositories to populate the web app in real time. Deployment only requires a static site-capable web server. The [Imageomics instance](https://catalog.imageomics.org) is currently deployed via GitHub Pages.
```

Author: Was there more you meant to write for this? (I ask because of the change to a comma at the end).

Reviewer: Typo. Fixed now.

> This is a template repository designed to be copied (with "Use this Template") and customized by different organizations to present their code, data, models, and spaces in a dynamic, searchable site.

Reviewer 2: Small wording suggestion: since this repository is both the Imageomics Catalog application and intended to be reusable as a template, could we clarify that distinction here? The current wording makes it sound like this is only a template repository

Author: This is a very good point, thank you! As the Imageomics catalog, it could have a PR to add additional GitHub or Hugging Face repos too.

Author: @beanbean9339, I updated the description in [c1d7703](https://github.com/Imageomics/catalog/pull/118/commits/c1d7703887a7798fe5d9c34755b4f26811e60889). I also added a note about contributing to the `README` in [7ceec95](https://github.com/Imageomics/catalog/pull/118/commits/7ceec959c4a61e4df1bd751381c66313eafeefe0).

> This is a template repository designed to be copied (with "Use this Template") and customized by different organizations to present their code, data, models, and spaces in a dynamic, searchable site. The default setup is for the Imageomics Institute's GitHub and Hugging Face products, which is hosted from this repo.

Reviewer: ```suggestion
This is a template repository designed to be copied (with "Use this Template") and customized by different organizations to present their code, data, models, and spaces in a dynamic, searchable site. The default setup is for the Imageomics Institute's GitHub and Hugging Face products.
```

Author: Makes sense with the above change.

## Getting Started

Follow [development guidance](README.md#development-prerequisites) for local setup and formatting expectations. 

### Testing Changes

Always test your changes locally with `npm test` before submitting a PR to ensure:

- Content renders correctly.
- Fetching, sorting, and filtering appear as intended.
- No build errors occur.

UI features should be tested locally through running a preview.

## Coding Style, Conventions, and Project Structure

> Please refer to our [AGENTS.md](AGENTS.md). This also includes special notes on key differences with the Hugging Face API and important considerations for the templated design of this repository.

Reviewer: _differences with_ normally invokes one thing being compared to another. The other presumably is the _Hugging Face API_, but it's semantically  not clear what the subject being compared with HF is. Perhaps spell out to _key differences between XX and the Hugging Face API_?

Author: Ahh yes. It's highlighting the differences in API calls between different Hugging Face repo types.

> Please refer to our [AGENTS.md](AGENTS.md). This also includes special notes on key differences with the Hugging Face API and important considerations for the templated design of this repository.

Author: ```suggestion
Please refer to our [AGENTS.md](AGENTS.md). This also includes special notes on key differences between the repo types (datasets, models, and spaces) in Hugging Face API calls and returns,  as well as important considerations for the templated design of this repository.
```
@hlapp, how's this?

Reviewer: Great, applied it!

## Contribution Process

> 1. **Create an issue** describing the change or problem **in your own words**.

Reviewer: ```suggestion
1. **Create an issue** describing the problem or necessary change **in your own words**.
```

    - Check existing [issues](https://github.com/Imageomics/catalog/issues) first.
    - If you'd like to work on an existing issue, please comment on it to express interest or describe your potential fix.
2. **Create a feature branch** from `dev`. 
    - It is important that a repo fork includes the `dev` branch content since updates are staged there.
3. **Make your changes** following the standards above.
4. **Test locally** with `npm test`
5. **Run linter/formatter** to ensure formatting consistency
   - See [formatting standard](README.md#formatting-standard)
6. **Submit a pull request** with:
   - Clear description of changes **in your own words**.
   - Reference to related issue.
   - Screenshots if UI changes are involved.

> ![IMPORTANT]
> Please describe your changes or issues in your own words, not with AI-generated text. We understand that AI is sometimes used to help refine or improve writing; this is fine. AI has been used collaboratively in developing this template, and similar use is acceptable, though **AI use in PRs must be acknowledged and checked by the PR author**.

### Pull Request Guidelines

- Keep PRs focused on a single topic when possible.

> - Included tests and documentation updates for changes, as needed.

Reviewer 2: ```suggestion
- Include tests and documentation updates for changes, as needed.
```
